index.js

```diff
@@ -22,13 +23,31 @@ function whichPolygon(data) {
 
     var tree = rbush().load(bboxes);
 
-    return function query(p) {
+    function query(p) {
         var result = tree.search([p[0], p[1], p[0], p[1]]);
         for (var i = 0; i < result.length; i++) {
             if (insidePolygon(result[i][4], p)) return result[i][5];
         }
         return null;
+    }
+
+    query.tree = tree;
+    query.bbox = function queryBbox(bbox) {
+        var result = tree.search(bbox).filter(function (el) {
+            // bbox is (possibly) completely inside the polygon. Keep it.
+            if (insidePolygon(el[4], [(bbox[0] + bbox[2]) / 2, (bbox[1] + bbox[3]) / 2])) return true;
+            // polygon crosses or is totally inside the bbox. Keep it.
+            // Todo: account for holes.
+            if (lineclip(el[4][0], bbox).length > 0) return true;
+            return false;
+        }).map(function (el) {
+            return el[5];
+        });
```

Comment:
Let's use a simple loop like in `query` instead of two filter/map closures for consistency.